src/main/kotlin/no/nav/familie/ks/sak/kjerne/behandling/steg/vilkårsvurdering/validering/BarnetsAlderVilkårValidator2021og2024.kt

```diff
@@ -0,0 +1,36 @@
+@file:Suppress("ktlint:standard:filename")
+
+package no.nav.familie.ks.sak.kjerne.behandling.steg.vilkårsvurdering.validering
+
+import no.nav.familie.ks.sak.common.tidslinje.IkkeNullbarPeriode
+import no.nav.familie.ks.sak.common.util.DATO_LOVENDRING_2024
+import no.nav.familie.ks.sak.common.util.sisteDagIMåned
+import no.nav.familie.ks.sak.kjerne.behandling.steg.vilkårsvurdering.VilkårLovverkInformasjonForBarn
+import no.nav.familie.ks.sak.kjerne.behandling.steg.vilkårsvurdering.domene.VilkårResultat
+import no.nav.familie.ks.sak.kjerne.personopplysninggrunnlag.domene.Person
+import org.springframework.stereotype.Component
+
+@Component
+class BarnetsAlderVilkårValidator2021og2024(
+    val barnetsAlderVilkårValidator2021: BarnetsAlderVilkårValidator2021,
+    val barnetsAlderVilkårValidator2024: BarnetsAlderVilkårValidator2024,
+) {
+    fun validerBarnetsAlderVilkår(
+        perioder: List<IkkeNullbarPeriode<VilkårResultat>>,
+        barn: Person,
+        vilkårLovverkInformasjonForBarn: VilkårLovverkInformasjonForBarn,
+        behandlingSkalFølgeNyeLovendringer2024: Boolean,
+    ): List<String> {
+        val barnErAdoptert = perioder.any { it.verdi.erAdopsjonOppfylt() }
+
+        if (!barnErAdoptert && perioder.size != 2) {
```

Comment:
Skulle vi fjerne `barnErAdoptert`?

Reply:
Og også endre feilmelding til noe ala "Vilkåret for barnets alder må splittes i to perioder fordi den strekker seg over lovendringen 01.08.2024. Periodene må være som følgende: < perioder her >"

Reply:
Enig!